src/parse.c

```diff
@@ -1463,8 +1463,15 @@ const mapping_entry_handler network_handlers[] = {
  * Grammar and handlers for root node
  ****************************************************/
 
+/* we don't stress about validating this as it only affects apply */
+static gboolean noop_handler(yaml_document_t* doc, yaml_node_t* node, const void* _, GError** error)
+{
+    return TRUE;
+}
+
 const mapping_entry_handler root_handlers[] = {
     {"network", YAML_MAPPING_NODE, NULL, network_handlers},
+    {"replug", YAML_MAPPING_NODE, noop_handler},
```

Comment:
As above; let's make sure all the parsing happens in a single place, preferrably in C -- you want this to be as fast as possible and with as few dependencies as possible, since it happens at early boot.

Reply:
How would doing this in C work? The C layer doesn't do replugging, and none of the files it spits out control that process. AFAICT it's only done when you call `netplan apply`, which is pure python and also - unless I'm missing something - not run on boot.